### ISSUE TITLE
Macro expansion in for trait items

### ISSUE THREAD

**Issue by josephDunne** [rust-lang/rust#34183]
Hi, I see that in `ast.rs` there is no expectation for macro items inside a trait:
```Rust
pub enum TraitItemKind {
    Const(P<Ty>, Option<P<Expr>>),
    Method(MethodSig, Option<P<Block>>),
    Type(TyParamBounds, Option<P<Ty>>),
}
```
Is there a reason for this? I would like to build up Trait definitions via macros the same way I would build a struct or impl items using macros.

**Comment by jseyfried** [comment]
c.f. #5668
cc @nrc -- do you think we should support this?

**Comment by josephDunne** [comment]
My use case is generating trait extensions for a builder objects for the Rust version of Flatbuffers: https://github.com/google/flatbuffers/pull/3894 
Would be nice to get rid of the boilerplate with macros.

**Comment by nrc** [comment]
I think we probably could, I'm in no great rush, but it seems like a good thing.

**Comment by jseyfried** [comment]
This was fixed in #34213.
